src/components/AugmentedResultList/AugmentedResultList.ts

```diff
@@ -0,0 +1,141 @@
+    /**
+     * Data to add to every result with matching object id.
+     */
+    commonData: {};
+}
+
+/**
+ * Generic interface for the response returned by the external fetch action.
+ *
+ * @export
+ * @interface IPromiseReturnArgs
+ * @template T
+ */
+export interface IPromiseReturnArgs<T> {
+    data: T;
+}
+
+export interface AugmentedResultListOptions extends IResultListOptions {
+    matchingIdField: IFieldOption;
+    fetchAugmentData: (objectIds: String[]) => Promise<IPromiseReturnArgs<IAugmentData>>;
+    matchingFunction: (augmentData: any, queryResult: IQueryResult) => boolean;
+}
+
+export class AugmentedResultList extends Coveo.ResultList implements IComponentBindings {
+    static ID = 'AugmentedResultList';
+
+    /**
+     * @componentOptions
+     */
+    static options: AugmentedResultListOptions = {
+        /**
+         * The field to be used as matching ID between augment data and result.
+         */
+        matchingIdField: ComponentOptions.buildFieldOption({
+            required: true,
+        }),
+        /**
+         * The function used to fetch extra result information.
+         */
+        fetchAugmentData: ComponentOptions.buildCustomOption<(objectIds: String[]) => Promise<IPromiseReturnArgs<IAugmentData>>>(() => {
+            return null;
+        }),
+        /**
+         * The function to use to determine a match between augment data and query results.
+         */
+        matchingFunction: ComponentOptions.buildCustomOption<(augmentData: any, queryResult: IQueryResult) => boolean>(() => {
+            return null;
+        }),
+    };
+
+    /**
+     * Creates a new `AugmentedResultList` component.
+     * @param element The HTMLElement on which to instantiate the component.
+     * @param options The options for the `ResultList` component.
+     * @param bindings The bindings that the component requires to function normally. If not set, these will be
+     * automatically resolved (with a slower execution time).
+     */
+    constructor(public element: HTMLElement, public options: AugmentedResultListOptions, public bindings: IComponentBindings) {
+        super(element, ComponentOptions.initComponentOptions(element, AugmentedResultList, options), bindings, AugmentedResultList.ID);
+        if (!this.options.matchingFunction) {
+            this.options.matchingFunction = this.defaultMatchingFunction;
+        }
+    }
+
+    private defaultMatchingFunction = (augmentData: any, queryResult: IQueryResult) => {
+        const fieldName = this.getFieldString(this.options.matchingIdField);
+        return augmentData[fieldName] === queryResult.raw[fieldName];
+    };
+
+    private getObjectPayload(results: IQueryResult[]): String[] {
+        const field = this.getFieldString(this.options.matchingIdField);
+        if (results.length > 0) {
+            return results.filter((result) => result.raw && result.raw[field]).map((result) => result.raw[field]);
+        }
+        return [];
```

Comment:
The if is redundant. You can simplify using
```suggestion
        return results.filter((result) => result.raw && result.raw[field]).map((result) => result.raw[field]);
```